data_training.ipynb

```diff
@@ -0,0 +1,305 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "RAND_STATE = 0\n",
+    "\n",
+    "X = ob_df.drop(columns=[\"NObeyesdad\"])\n",
+    "y = ob_df[\"NObeyesdad\"]\n",
+    "\n",
+    "# Split the data\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, train_size=0.7, stratify=y, random_state=RAND_STATE)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "818226ed-11d9-433a-ac55-e6736c7c82a5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_train"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f9268f75-35a0-4d05-9e04-f93f3d593266",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_test"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f1834968-a6e7-49db-9048-2cc97c5478f3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_train"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2e9775ac-8cb4-4258-b906-233d75ef3dfb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5fc323c2-3566-404c-8ef1-ae6084e1b28f",
+   "metadata": {},
+   "source": [
+    "### Feature Scaling"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a278e736-596e-4fe5-9c1e-1c5816387a0e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "numerical_features = [\"Age\", \"Height\", \"Weight\", \"FCVC\", \"NCP\", \"CH2O\", \"FAF\", \"TUE\"]\n",
+    "\n",
+    "# Fit the scaler to the train set, not the test set, to prevent \"leaking\" of the test set.\n",
+    "scaler = StandardScaler().fit(X_train[numerical_features])\n",
+    "\n",
+    "X_train[numerical_features] = scaler.transform(X_train[numerical_features])\n",
+    "X_test[numerical_features] = scaler.transform(X_test[numerical_features])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "105a6d38-a058-4cc0-9c55-655adf9da64b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_train"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42669e1c-bb68-40d6-8729-f99fc24ada22",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X_test"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "41531b10-a447-44f0-b6a6-e0ffcfd47737",
+   "metadata": {},
+   "source": [
+    "### Categorical Feature Encoding"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b4187423-a69e-46c3-a329-cb5ff18cf639",
+   "metadata": {},
+   "source": [
+    "#### Target Feature"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4866270c-96df-429a-bdcb-bf34028bd4cf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "le = LabelEncoder()\n",
+    "le.fit(ob_df[\"NObeyesdad\"])\n",
+    "\n",
+    "y_train[:] = le.transform(y_train)\n",
+    "y_test[:] = le.transform(y_test)"
+   ]
+  },
```

Comment:
- Label encoding is appropriately used for the target variable. 
- Ensure the `y_train[:]` and `y_test[:]` correctly reference and update the series; alternatively, consider using `.loc` for clarity.